Jim-Hodapp-Coaching/refactor-platform-fe
# 17 fix auth store refresh page bug

## Description
Updates the useEffect function to fix the infinite loop occurring on page refresh. 

#### GitHub Issue: [Fixes] #17 

### Changes
* Added userUUID to the dependency array of useEffect. This ensures the effect only runs when userUUID changes.
* Added a check in loadOrganizations to prevent fetching if userUUID is not available. This avoids unnecessary calls on initial render.
* Includes the changes by Jim from #14 making that PR no longer necessary.

### Screenshots / Videos Showing UI Changes (if applicable)


### Testing Strategy
- Login with valid credentials
- Refresh the dashboard
- Confirm infinite loop is absent by checking the console for errors

### Concerns
_describe any concerns that might be worth mentioning or discussing_
